packages/flutter_tools/lib/src/build_system/targets/hook_runner_native.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../asset.dart';
+import '../../build_info.dart';
+import '../../globals.dart' as globals;
+import '../../hook_runner.dart' show FlutterHookRunner;
+import '../../isolated/native_assets/dart_hook_result.dart' show DartHooksResult;
+import '../build_system.dart' show BuildResult, Environment, ExceptionMeasurement;
+import 'native_assets.dart' show DartBuild;
+
+class FlutterHookRunnerNative implements FlutterHookRunner {
+  FlutterHookResult? _flutterHookResult;
+
+  @override
+  Future<FlutterHookResult> runHooks({
+    required TargetPlatform targetPlatform,
+    required Environment environment,
+  }) async {
+    globals.printTrace('runDartBuild() with ${environment.defines} and $targetPlatform');
```

Comment:
We're trying to slowly reduce the number of sites where we're accessing `globals` within the tool. This class should have a `Logger` instance passed in instead.